modules/react/modal/lib/ModalOverlay.tsx

```diff
@@ -83,6 +84,7 @@ const OpenModalOverlay = createSubcomponent('div')({
   elemPropsHook: useModalOverlay,
 })<ModalOverlayProps>((elemProps, Element, model) => {
   const windowSize = useWindowSize();
+  const theme = useTheme();
```

Comment:
Curious about this. The way I tested this was going to this [file](https://github.com/Workday/canvas-kit/blob/master/modules/react/modal/stories/stories_VisualTesting.tsx#L60) and instead of using the `useTheme` hook, i just moved the contents of `useTheme` inside of the `them` prop where `CanvasProivder` is being used. I was able to reproduce the error saying that  `uncaught TypeError: Cannot read properties of undefined (reading 'down') `.  Adding this here and keeping that as a test seemed to remove this error but either way I added documentation around how to work around this in the upgrade guide.

Reply:
`useTheme` is terribly inefficient as-is. We need to fix it. Or replace it.

I recommend making a `useCanvasTheme` and a `getCanvasTheme` and use that internally and deprecate `useTheme` and remove it in v9.

Reply:
See https://github.com/Workday/canvas-kit/issues/1899#issuecomment-1303911159